horreum-web/src/components/LabelFilter/LabelFilterOption.tsx

```diff
@@ -0,0 +1,111 @@
+    filter: any;
+    setFilter(filter: any): any;
+    selection: string;
+    setSelection(selection: string): any;
+}
+
+
+export default function LabelFilterOption(props: FilterOptionProps) {
+    const [isMenuOpen, setIsMenuOpen] = React.useState<boolean>(false);
+    const toggleRef = React.useRef<HTMLButtonElement>(null);
+    const menuRef = React.useRef<HTMLDivElement>(null);
+    const containerRef = React.useRef<HTMLDivElement>(null);
+
+    const handleMenuKeys = (event: KeyboardEvent) => {
+        if (isMenuOpen && menuRef.current?.contains(event.target as Node)) {
+            if (event.key === 'Escape' || event.key === 'Tab') {
+                setIsMenuOpen(!isMenuOpen);
+                toggleRef.current?.focus();
+            }
+        }
+    };
+
+    const handleClickOutside = (event: MouseEvent) => {
+        if (isMenuOpen && !menuRef.current?.contains(event.target as Node)) {
+            setIsMenuOpen(false);
+        }
+    };
+
+    React.useEffect(() => {
+        window.addEventListener('keydown', handleMenuKeys);
+        window.addEventListener('click', handleClickOutside);
+        return () => {
+            window.removeEventListener('keydown', handleMenuKeys);
+            window.removeEventListener('click', handleClickOutside);
+        };
+    }, [isMenuOpen, menuRef]);
+
+    const onToggleClick = (ev: React.MouseEvent) => {
+        ev.stopPropagation(); // Stop handleClickOutside from handling
+        setTimeout(() => {
+            if (menuRef.current) {
+                const firstElement = menuRef.current.querySelector('li > button:not(:disabled)');
+                firstElement && (firstElement as HTMLElement).focus();
+            }
+        }, 0);
+        setIsMenuOpen(!isMenuOpen);
+    };
+
+    function onSelect(event: React.MouseEvent | undefined, itemId: string | number | undefined) {
+        if (typeof itemId === 'undefined') {
+            return;
+        }
+
+        props.setSelection(itemId.toString());
+        const newFilter = {...props.filter};
+        newFilter[props.name] = itemId;
+        props.setFilter(newFilter);
+        setIsMenuOpen(!isMenuOpen);
+    }
+
+    const toggle = (
+        <MenuToggle
+            ref={toggleRef}
+            onClick={onToggleClick}
+            isExpanded={isMenuOpen}
+            style={
+                {
+                    width: '200px'
+                } as React.CSSProperties
+            }
+        >
+            Filter by {props.name}
+        </MenuToggle>
+    );
+
+    const menu = (
+        <Menu ref={menuRef} id="attribute-search-{props.name}-menu" onSelect={onSelect} selected={props.selection}>
+            <MenuContent>
+                <MenuList>
+                    {
+                        props.options.map((option) => (
+                            <MenuItem isSelected={props.selection === option} itemId={option} key={option}>
+                                {option}
+                            </MenuItem>
```

Comment:
We need to filter out `undefined` options otherwise we would have null options (that are selected by default):
![image](https://github.com/Hyperfoil/Horreum/assets/26715795/ed36f920-b2e1-47e1-beb7-fb12a57a76c9)
```suggestion
                        props.options.filter(option => option).map((option) => (
                            <MenuItem isSelected={props.selection === option} itemId={option} key={option}>
                                {option}
                            </MenuItem>
```

Reply:
The filter does remove out `undefined`, but the api (maybe it is the json marshalling) is returning a mixture of `null` and undefined, and it is the `null` values that are causing the issue. 

Other LabelValues only return `undefined` for an unknown value.

My intention was to merge this PR as it is, and then fix the api problem that is returning the wrong data type for an unknown label value

Reply:
Got it :+1: